From 7db0af8abff5976f28880746075417f32236b694 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Sat, 4 Oct 2025 02:20:03 +0300
Subject: [PATCH 3/3] freedreno/a5xx: skip NULL resources in fd5_launch_grid()

The global_bindings can contain NULL entries. Skip those while counting
entries and writing the CP_NOP packet.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/37626]
---
 src/gallium/drivers/freedreno/a5xx/fd5_compute.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/gallium/drivers/freedreno/a5xx/fd5_compute.c b/src/gallium/drivers/freedreno/a5xx/fd5_compute.c
index 19f9b4de9bd2..b29c5a3ec1f1 100644
--- a/src/gallium/drivers/freedreno/a5xx/fd5_compute.c
+++ b/src/gallium/drivers/freedreno/a5xx/fd5_compute.c
@@ -107,7 +107,8 @@ fd5_launch_grid(struct fd_context *ctx,
    fd5_emit_cs_consts(v, ring, ctx, info);
 
    util_dynarray_foreach (&ctx->global_bindings, struct pipe_resource *, res)
-      nglobal++;
+      if (*res)
+         nglobal++;
 
    if (nglobal > 0) {
       /* global resources don't otherwise get an OUT_RELOC(), since
@@ -118,7 +119,8 @@ fd5_launch_grid(struct fd_context *ctx,
        */
       OUT_PKT7(ring, CP_NOP, 2 * nglobal);
       util_dynarray_foreach (&ctx->global_bindings, struct pipe_resource *, res)
-         OUT_RELOC(ring, fd_resource(*res)->bo, 0, 0, 0);
+         if (*res)
+            OUT_RELOC(ring, fd_resource(*res)->bo, 0, 0, 0);
    }
 
    const unsigned *local_size =
-- 
2.51.0

